subscription_gateway.go

```diff
@@ -53,3 +56,33 @@ func (g *SubscriptionGateway) Cancel(ctx context.Context, subId string) (*Subscr
 	}
 	return nil, &invalidResponseError{resp}
 }
+
+// RetryCharge retries to charge for a subscription. The amount has to
+// be > 0.
+func (g *SubscriptionGateway) RetryCharge(ctx context.Context, subId string, amount Decimal) error {
```

Comment:
We should give the user more control over the transaction that gets created out of the retry. For example in the Braintree Java SDK there are four functions:

https://github.com/braintree/braintree_java/blob/8fa6a13/src/main/java/com/braintreegateway/SubscriptionGateway.java#L108-L132

```java
public Result<Transaction> retryCharge(String subscriptionId) { ... }
public Result<Transaction> retryCharge(String subscriptionId, BigDecimal amount) { ... }
public Result<Transaction> retryCharge(String subscriptionId, Boolean submitForSettlement) { ... }
public Result<Transaction> retryCharge(String subscriptionId, BigDecimal amount, Boolean submitForSettlement) { ... }
```

I think we should probably create an exported struct `SubscriptionTransactionRequest` like what the Braintree Java SDK has, and have that be the only data the user needs to provide. e.g. https://github.com/braintree/braintree_java/blob/8fa6a13/src/main/java/com/braintreegateway/SubscriptionTransactionRequest.java